BLSQ/iaso
# Polio 570 xls form generation

New endpoint at /api/polio/campaigns/generate_xlsform/ that allows to export a xlsform that contains the data of a specific campaign. 


## Self proof reading checklist

## Changes

It will export by default some fixed data as notes : 
OBR Name
Round Numbers
First round date
Vaccine type

As selectable field:
All org units from the campaign scope.

As it's the description of the campaign and must be at the top of the xlsform.

You can query on authorised fields (Non FK Fields) of a campaign and insert it automatically into the xlsfile by naming your variable starting by "insert_".  In this example the field {insert_doses_requested} will have the count of the doses from the campaign. Once the xlsform generated the number of doses will be added into the calculation column and will be available for calculation by other rows. As you can see at row 30 in note1. 
![image](https://user-images.githubusercontent.com/45785235/192313971-ddd5a42c-d39b-404a-be9e-7536b3524c86.png)


## How to test

Explain how to test your PR.
To test it you need a xls file formated as xlsform. 
If you want to extract data from campaign and insert it into the xlsform you must do as follow : 
![image](https://user-images.githubusercontent.com/45785235/192316022-231566c5-fc48-4547-87c3-7ef41283cb1f.png)
The data will be added in the calculation column.

You must upload the form template at /api/polio/campaignformtemplate

Export your generated xlsform from the template at /api/polio/campaigns/generate_xlsform/
it must contain the query param uuid campaign and template_name as follow : 
api/polio/campaigns/generate_xlsform/?id={campaign.uuid}&form_name={campaignformtemplate.name}

The form will be then ready to use in enketo

You can check this video to see how it works ( I upload via admin on the video )


## Print screen / video


https://user-images.githubusercontent.com/45785235/192324180-8403fd5b-05b1-4afb-8e68-38bc0c5d416b.mov


## Note : 

Once the campaigns will be multi tenancy, we must ensure that we can export forms only from campaigns that are accessible for the user.
